specification/Metadata/0.0.0/README.md

```diff
@@ -0,0 +1,1217 @@
+          "name": "Previous 5 sensor readings",
+          "description": "This example demonstrates the use of `[]` for null arrays as well as how `null` can be used for individual array components.",
+          "type": "ARRAY",
+          "componentType": "STRING",
+          "optional": true,
+        },
+        "condition": {
+          "name": "Sensor condition",
+          "description": "This example shows how to use `default` to fill in `null` elements. In this case, `null` means `Good Condition`",
+          "type": "STRING",
+          "optional": true,
+          "default": "Good Condition"
+        }
+      }
+    }
+  },
+  "instanceTables": {
+    "sensorTable": {
+      "class": "sensor",
+      "count": 3,
+      "properties": {
+        "currentTemperature": {
+          "values": [
+            0.9,
+            0.8,
+            null
+          ]
+        },
+        "history": {
+          "values": [
+            [0.8, 0.9, 0.9, 0.8, 0.8],
+            [null, null, 0.7, 0.6, 0.7],
+            []
+          ]
+        },
+        "condition": {
+          "values": [
+            null,
+            null,
+            "Malfunctioning. Needs maintenance"
+          ]
+        }
+      }
+    }
+  }
+}
+```
+
+#### Single Instance Shorthand
+
+In most situations, there are several instances of a class. Therefore, an array is a natural choice for the JSON representation. In cases where only a single instance is used, this specification allows for a simpler property dictionary.
+
+To enable this shorthand, omit the `count` field. Then `properties` is interpreted as a dictionary of property names to single values. This is more convenient than the usual behavior where a nested object is used for each property.
+
+The following example compares the two formats of JSON encoding:
+
+```json
+{
+  "classes": {
+    "datasetInfo": {
+      "properties": {
+        "date": {
+          "type": "STRING",
+        },
+        "authors": {
+          "type": "ARRAY",
+          "componentType": "STRING",
+        },
+        "company": {
+          "type": "STRING",
+          "optional": true
+        }
+      }
+    }
+  },
+  "instanceTables": {
+    "verboseTable": {
+      "name": "Verbose property example",
+      "description": "This is the typical way to represent properties, as it supports any number of instances.",
+      "class": "datasetInfo",
+      "count": 1,
+      "properties": {
+        "date": {
+          "values": ["2020-11-03"]
+        },
+        "authors": {
+          "values": [
+            ["Jane Doe", "John Smith"]
+          ]
+        },
+        "company": {
+          "values": [null]
+        }
+      } 
+    },
+    "shorthandTable": {
+      "name": "Shorthand property example",
+      "description": "This is equivalent to the table above, just more concise",
+      "class": "datasetInfo",
+      "properties": {
+        "date": "2020-11-03",
+        "authors": ["Jane Doe", "John Smith"]
```

Comment:
For clarity the summary above should say that optional properties may be omitted in the short-hand version